src/Common/src/System/Security/Cryptography/Asn1/Pbkdf2Params.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.Security.Cryptography.Asn1
+{
+    // https://tools.ietf.org/html/rfc2898#appendix-A.2
+    //
+    // PBKDF2-params ::= SEQUENCE {
+    //   salt CHOICE {
+    //     specified OCTET STRING,
+    //     otherSource AlgorithmIdentifier {{PBKDF2-SaltSources}}
+    //   },
+    //   iterationCount INTEGER (1..MAX),
+    //   keyLength INTEGER (1..MAX) OPTIONAL,
+    //   prf AlgorithmIdentifier {{PBKDF2-PRFs}} DEFAULT
+    //   algid-hmacWithSHA1
+    // }
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct Pbkdf2Params
+    {
+        public Pbkdf2SaltChoice Salt;
+
+        // The spec calls out that while there's technically no limit to IterationCount,
+        // that specific platforms may have their own limits.
+        //
+        // This defines ours to uint.MaxValue (and, conveniently, not a negative number)
+        public uint IterationCount;
+
+        // The biggest value that makes sense currently is 256/8 => 32.
+        [OptionalValue]
+        public byte? KeyLength;
+
+        [DefaultValue(
```

Comment:
some comment what this default value means would be useful

Reply:
it's in the ASN description as the struct comment.  Though apparently with a weird newline.  It's algid-hmacWithSHA1